snuba/datasets/spans_processor.py

```diff
@@ -0,0 +1,129 @@
+import uuid
+from datetime import datetime
+from typing import Any, List, Mapping, MutableMapping, Optional, Tuple
+
+from sentry_relay.consts import SPAN_STATUS_NAME_TO_CODE
+
+from snuba.datasets.events_format import enforce_retention, extract_extra_tags
+from snuba.processor import (
+    InsertBatch,
+    MessageProcessor,
+    ProcessedMessage,
+    _as_dict_safe,
+    _ensure_valid_date,
+    _unicodify,
+)
+
+UNKNOWN_SPAN_STATUS = 2
+
+
+class SpansMessageProcessor(MessageProcessor):
+    def __extract_timestamp(self, field: float) -> Tuple[datetime, int]:
+        timestamp = _ensure_valid_date(datetime.fromtimestamp(field))
```

Comment:
I think I will wait for a solution to https://github.com/getsentry/snuba/pull/1336 before changing this, otherwise we would have a different timestamp between transactions and spans.